### PR TITLE
Implement issue 45   create config file

### LINKED ISSUE #45: Move 'UTF-8 encoding' and 'NoGUID' options from the user interface into a config file
Also add an option to the user interface to make this change for the entire repository to allow transition in one commit. Once transitioned the tool will pick up the setting from the config file.

### PR DESCRIPTION
**Pull request by bkp7**
Closes #45

### CODE REVIEW

File: Modules/modExtendSaveAsText.ACM [hunk 1]
```diff
@@ -27,7 +27,7 @@ Private Type AccObject
   blnAppended As Boolean 'Indicating whether or not ithe object has been appended to its parent yet
 End Type
 
-Private Type Ignore
+Private Type IGNORE
```
Comment (gilbertbw) [inline Modules/modExtendSaveAsText.ACM]: Is this capilised for a reason? I this Reddick convention would be `Ignore`

File: Modules/modExtendSaveAsText.ACM [hunk 2]
```diff
@@ -36,21 +36,9 @@ Private Const clngNOTFOUND = -2147483648#
...
-' @version 1.0
-'************************************************
-
-Public Sub SaveAsText(objType As asObjectType, objName As String, strFilename As String)
-  Dim Ignores(7) As Ignore
-  Dim blnSaveAsUTF8 As Boolean
-  Dim blnSourceIsUCS2 As Boolean
-  Dim blnFail As Boolean
-
+Private Sub PopIgnoreArr(ByRef Ignores() As IGNORE)
```
Comment (gilbertbw) [inline Modules/modExtendSaveAsText.ACM]: nit: Rename to `PopulateIgnoreArr`? I thought this popped the first element off based on the name

File: Modules/modExtendSaveAsText.ACM [hunk 3]
```diff
@@ -1126,7 +1134,67 @@ Private Sub AppendParent(AccObj() As AccObject)
...
+      CleanFormReportFileEx = ConvFileUCS2_to_UTF8(strFilenameNew)
+    Case "UCS2"
+      CleanFormReportFileEx = True
+    Case "ANSI":
+      CleanFormReportFileEx = ConvFileUCS2_to_ANSI(strFilenameNew)
+  End Select
+  
+End Function
+Private Function CleanFormReportFile(strFilename As String, strFilenameNew As String, Ignores() As IGNORE) As Boolean
   ' This function takes a file and cleans out the lines (Sections) specified in the Ignores Array
```
Comment (gilbertbw) [inline Modules/modExtendSaveAsText.ACM]: nit: meantion that the file has to be UCS2 encoded?

File: Forms/Access SVN - Options.ACF [hunk 4]
```diff
@@ -2512,6 +2607,36 @@ Private Sub cmdRun_Click()
...
+      Next
+    End If
+  Next
+  
+  SetConfigValue "UTF-8 Encoding", True, appendSlash(Me.txtOutputLocation.Value) & gstrConfigFilename
+    
+  Me.cmdRun.SetFocus
+  Me.cmdUTF8Encoding.Enabled = False
+  DoCmd.Hourglass False
+End Sub
```
[the remark this answers is not in the document]
Comment (bkp7) [reply Forms/Access SVN - Options.ACF]: Done

File: Forms/Access SVN - Options.ACF [hunk 5]
```diff
@@ -2496,6 +2580,17 @@ Private Sub cmdFolderDialog_Click()
...
+Private Sub cmdRemoveGUIDs_Click()
+  
+  DoCmd.Hourglass True
+  If CleanFormsReports(Me.txtOutputLocation.Value) Then
+    SetConfigValue "Remove GUIDs", True, appendSlash(Me.txtOutputLocation.Value) & gstrConfigFilename
+  End If
+  Me.cmdRun.SetFocus
+  Me.cmdRemoveGUIDs.Enabled = False
+  DoCmd.Hourglass False
+End Sub
```
[the remark this answers is not in the document]
Comment (bkp7) [reply Forms/Access SVN - Options.ACF]: Done

File: Forms/Access SVN - Options.ACF [hunk 6]
```diff
@@ -2512,6 +2607,36 @@ Private Sub cmdRun_Click()
...
+Private Sub cmdUTF8Encoding_Click()
+  Dim i As Long
+  Dim j As Long
+  Dim strFoldername As String
+  Dim strFilename As String
+  Dim arrSVNItems() As SVNITEM
+  
+  DoCmd.Hourglass True
+  For i = 0 To 1
+    strFoldername = appendSlash(Me.txtOutputLocation.Value) & IIf(i = 0, "Forms", "Reports")
```
[the remark this answers is not in the document]
Comment (bkp7) [reply Forms/Access SVN - Options.ACF]: Done

File: Forms/Access SVN - Options.ACF [hunk 7]
```diff
@@ -2581,11 +2706,13 @@ End Sub
...
   Me.txtLastImported = asGetLatestImportDate()
   Me.txtSvnUrl = SCC_GetMappedUrl(Me.txtOutputLocation.Value)
-If LenB(Nz(Me.txtSvnUrl.Value, vbNullString)) = 0 Or Left$(Me.txtSvnUrl.Value, 10) = "No mapped " Then ' No mapped url, cannot import, but allow export for original export
+  If LenB(Nz(Me.txtSvnUrl.Value, vbNullString)) = 0 Or Left$(Me.txtSvnUrl.Value, 10) = "No mapped " Then ' No mapped url, cannot import, but allow export for original export
     Me.btnImport.Enabled = False ' should probably trap for when the button has focus but cannot see how user could get to that point
   Else
     Me.btnImport.Enabled = True
   End If
+  Me.cmdUTF8Encoding.Enabled = Not GetConfigValue("UTF-8 Encoding", appendSlash(Me.txtOutputLocation.Value) & gstrConfigFilename)
+  Me.cmdRemoveGUIDs.Enabled = Not GetConfigValue("Remove GUIDs", appendSlash(Me.txtOutputLocation.Value) & gstrConfigFilename)
```
[the remark this answers is not in the document]
Comment (bkp7) [reply Forms/Access SVN - Options.ACF]: Ta

File: Forms/Access SVN - Options.ACF [hunk 8]
```diff
@@ -2512,6 +2607,36 @@ Private Sub cmdRun_Click()
...
+  For i = 0 To 1
+    strFoldername = appendSlash(Me.txtOutputLocation.Value) & IIf(i = 0, "Forms", "Reports")
+      
+    SCC_GetAllVersionsSubFolder strFoldername, arrSVNItems
+    
+    If (Not arrSVNItems) <> -1 Then ' check that the arr is not nothing before looping through it
+      For j = LBound(arrSVNItems) To UBound(arrSVNItems)
+        If arrSVNItems(j).svName <> "" Then
+          strFilename = appendSlash(strFoldername) & arrSVNItems(j).svName
+          ConvFile_to_UTF8 strFilename
```
[the remark this answers is not in the document]
Comment (bkp7) [reply Forms/Access SVN - Options.ACF]: Done

File: Modules/modConfig.ACM [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+    Case vbDecimal:
+    Case vbByte:
+    'Case vbLongLong
+    Case vbUserDefinedType:
+    Case vbArray:
+  End Select
+End Function
+
+Private Function FromString(str As String, typ As Integer) As Variant
+  ' Takes a string and a desired type and converets the string to a variant of that type
```
[the remark this answers is not in the document]
Comment (bkp7) [reply Modules/modConfig.ACM]: Done

File: Modules/modExtendSaveAsText.ACM (hunk 1 above)
Comment (bkp7) [reply Modules/modExtendSaveAsText.ACM]: Changed

File: Modules/modExtendSaveAsText.ACM (hunk 2 above)
Comment (bkp7) [reply Modules/modExtendSaveAsText.ACM]: Done

File: Modules/modExtendSaveAsText.ACM (hunk 3 above)
Comment (bkp7) [reply Modules/modExtendSaveAsText.ACM]: Done

File: Modules/modConfig.ACM [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+End Function
+
+Private Function GetDefaultValueType(strItem As String) As Integer
+  ' Returns the type of the key which is passed in (VarType)
+  Dim i As Long
+  
+  Call InitiateDefaults ' to ensure they are populated
+  
+  For i = LBound(marrDefaults) To UBound(marrDefaults)
+    If marrDefaults(i).Key = strItem Then
```
[the remark this answers is not in the document]
Comment (bkp7) [reply Modules/modConfig.ACM]: Have left as is as it has been tested already

File: Forms/Access SVN - Options.ACF [hunk 11]
```diff
@@ -2496,6 +2580,18 @@ Private Sub cmdFolderDialog_Click()
   Call txtOutputLocation_AfterUpdate
 End Sub
 
+Private Sub cmdRemoveGUIDs_Click()
+  
+  DoCmd.Hourglass True
+  If CleanFormsReports(Me.txtOutputLocation.Value) Then
+    SetConfigValue "Remove GUIDs", True, appendSlash(Me.txtOutputLocation.Value) & gstrConfigFilename
+  End If
```
Comment (gilbertbw) [inline Forms/Access SVN - Options.ACF]: Should the user be warned if this fails, as in the reencoding?

Comment (bkp7) [reply Forms/Access SVN - Options.ACF]: Done

File: Modules/modConfig.ACM [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+
+Private Const APPNAME As String = "MSAccess SCC Add-in"
+
+Private Type ConfigItem
+  Key As String
+  DefaultVal As Variant
+  VarType As Integer
+End Type
+
+Private marrDefaults() As ConfigItem
```
[the remark this answers is not in the document]
Comment (bkp7) [reply Modules/modConfig.ACM]: Done

**Pull request merged**